src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/DependencyAnalysis/Target_Wasm/UnboxingStubTargetMappingNode.cs

```diff
@@ -0,0 +1,142 @@
+            _unboxingStubNode = unboxingStubNode;
+        }
+
+        public IMethodNode Stub => _unboxingStubNode;
+        public ISymbolNode GetTarget(NodeFactory factor) => GetUnboxingStubTarget(factor, Stub);
+
+        protected override string GetName(NodeFactory factory) => $"Target mapping for {_unboxingStubNode.GetMangledName(factory.NameMangler)}";
+
+        protected override void OnMarked(NodeFactory factory) => factory.UnboxingStubTargetMappings.AddMapping(this);
+
+        public override bool StaticDependenciesAreComputed => true;
+        public override IEnumerable<DependencyListEntry> GetStaticDependencies(NodeFactory context) => null;
+
+        public override bool InterestingForDynamicDependencyAnalysis => false;
+        public override bool HasDynamicDependencies => false;
+        public override IEnumerable<CombinedDependencyListEntry> SearchDynamicDependencies(List<DependencyNodeCore<NodeFactory>> markedNodes, int firstNode, NodeFactory context) => null;
+
+        public override bool HasConditionalStaticDependencies => false;
+        public override IEnumerable<CombinedDependencyListEntry> GetConditionalStaticDependencies(NodeFactory context) => null;
+
+        public override int ClassCode => 1371471006;
+
+        public override int CompareToImpl(ISortableNode other, CompilerComparer comparer)
+        {
+            // We rely on sorting regular unboxing stubs before instantiating ones.
+            IMethodNode stub = Stub;
+            IMethodNode otherStub = ((UnboxingStubTargetMappingNode)other).Stub;
+            if (stub.GetType() != otherStub.GetType())
+            {
+                return stub is UnboxingStubNode ? -1 : 1;
+            }
+
+            return comparer.Compare(stub, otherStub);
+        }
+
+        public static void AddConditionalMappingDependency(ref CombinedDependencyList dependencies, NodeFactory factory, IMethodNode unboxingStubNode)
+        {
+            // The mapping is only needed if the target will actually be present in the output.
+            if (factory.UseUnboxingStubTargetMappings && GetUnboxingStubTarget(factory, unboxingStubNode) is ISymbolNode target)
+            {
+                dependencies ??= new();
+                UnboxingStubTargetMappingNode mapping = factory.UnboxingStubTargetMapping(unboxingStubNode);
+                dependencies.Add(new(mapping, target, "Unboxing stub target mapping"));
+            }
+        }
+
+        private static ISymbolNode GetUnboxingStubTarget(NodeFactory factory, IMethodNode stub) => stub switch
+        {
+            UnboxingStubNode simple => simple.GetUnderlyingMethodEntrypoint(factory),
+            ISpecialUnboxThunkNode { IsSpecialUnboxingThunk: true } instantiating => instantiating.GetUnboxingThunkTarget(factory),
+            _ => null
+        };
+    }
+
+    public sealed class UnboxingStubTargetMappingsNode : EmbeddedDataContainerNode
```

Comment:
Arguably `UnboxingStubTargetMappingsNode` could live in its own file... I think the ILC style so has been to pack related classes like these together, however.